src/Geometry.cpp

```diff
@@ -3938,6 +3936,8 @@ void Geometry::loadFromFile(std::string filename, bool non_uniform_lattice,
     else {
       log_printf(ERROR, "Unsupported surface type %s", name);
     }
+    if (name != "")
+      delete [] name;
```

Comment:
For simplicity, we could keep only one in name and str. To avoid the warning, we could use name = (char*)""